security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -406,16 +406,17 @@ func (s *sdsservice) clearStaledClients() {
 func NotifyProxy(connKey cache.ConnKey, secret *model.SecretItem) error {
 	conIDresourceNamePrefix := sdsLogPrefix(connKey.ConnectionID, connKey.ResourceName)
 	sdsClientsMutex.Lock()
-	defer sdsClientsMutex.Unlock()
 	conn := sdsClients[connKey]
 	if conn == nil {
+		sdsClientsMutex.Unlock()
```

Comment:
any issue if using defer unlock ?

Reply:
The defer unlock will include channel push in the lock period, which unnecessary holds the lock especially if the channel is busy.  It is just generally not desirable to do so.